src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/MethodTypeInference.cs

```diff
@@ -97,6 +97,12 @@ private enum Dependency
         private Dependency[,] _dependencies; // Initialized lazily
         private bool _dependenciesDirty;
 
+        /// <summary>
+        /// For error recovery, we allow a mismatch between the number of arguments and parameters
+        /// during type inference. This sometimes enables inferring the type for a lambda parameter.
+        /// </summary>
+        private int Length => System.Math.Min(_arguments.Length, _formalParameterTypes.Length);
```

Comment:
AmountOfArgumentsToConsider?

---
In reply to: [66173546](https://github.com/dotnet/roslyn/pull/11807#discussion_r66173546) [](ancestors = 66173546)